tests/fv/fv_test.go

```diff
@@ -0,0 +1,90 @@
+package fv_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/projectcalico/k8s-policy/tests/testutils"
+
+	"github.com/projectcalico/felix/fv/containers"
+	"github.com/projectcalico/k8s-policy/tests/testutils"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+)
+
+const kubeconfigTemplate = `apiVersion: v1
+kind: Config
+clusters:
+- name: test 
+  cluster:
+    server: http://%s:8080
+users:
+- name: calico
+contexts:
+- name: test-context
+  context:
+    cluster: test  
+    user: calico
+current-context: test-context`
+
+var _ = Describe("PolicyController", func() {
+	var (
+		etcd             *containers.Container
+		policyController *containers.Container
+		apiserver        *containers.Container
+		etcdClient       *client.Client
+		k8sClient        *kubernetes.Clientset
+	)
+
+	BeforeEach(func() {
+		// Run etcd.
+		etcd = testutils.RunEtcd()
+		etcdClient = GetEtcdClient(etcd.IP)
+		err := etcdClient.EnsureInitialized()
+		Expect(err).NotTo(HaveOccurred())
+
+		// Run apiserver.
+		apiserver = testutils.RunK8sApiserver(etcd.IP)
+
+		// Write out a kubeconfig file
+		kfconfigfile, err := ioutil.TempFile("", "ginkgo-policycontroller")
+		defer os.Remove(kfconfigfile.Name())
+		data := fmt.Sprintf(kubeconfigTemplate, apiserver.IP)
+		kfconfigfile.Write([]byte(data))
+
+		policyController = RunPolicyController(etcd.IP, apiserver.IP, kfconfigfile.Name())
+
+		k8sClient, err = GetK8sClient(kfconfigfile.Name())
+
+		// TODO: Use upcoming port checker functions to wait until apiserver is responding to requests.
+		time.Sleep(time.Second * 15)
+	})
+
+	It("should create profiles in etcd to represent k8s namespaces", func() {
+		// Create the namespace in Kubernetes
+		ns := &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "peanutbutter",
+			},
+			Spec: v1.NamespaceSpec{},
+		}
+		_, err := k8sClient.CoreV1().Namespaces().Create(ns)
+		Expect(err).NotTo(HaveOccurred())
+
+		// TODO: retry instead of waiting
+		time.Sleep(3 * time.Second)
```

Comment:
You could use a gomega `Eventually` call to replace this pretty easily.

https://onsi.github.io/gomega/#eventually

Reply:
Can I leave as is until Neil's `tryConnect` function goes in?

https://github.com/projectcalico/felix/pull/1548/files#diff-e7ba074d9458cbcba06250a59dfd889bR76

Reply:
I think we need the eventually functionality here anyway - we don't know how long it will take for the policy controller to write the data to etcd.

It's only a couple of lines and makes the tests much more robust.